lightningd/onion_message.c

```diff
@@ -197,8 +222,62 @@ void handle_obs_onionmsg_forward(struct lightningd *ld, const u8 *msg)
 
 void handle_onionmsg_to_us(struct lightningd *ld, const u8 *msg)
 {
-	/* FIXME! */
-	return;
+	struct onion_message_hook_payload *payload;
+	u8 *submsg;
+	struct secret *self_id;
+	size_t submsglen;
+	const u8 *subptr;
+
+	payload = tal(ld, struct onion_message_hook_payload);
+	payload->obsolete = false;
+	payload->om = tlv_onionmsg_payload_new(payload);
+	payload->blinding_in = NULL;
+	payload->our_alias = tal(payload, struct pubkey);
+
+	if (!fromwire_gossipd_got_onionmsg_to_us(payload, msg,
+						 payload->our_alias,
+						 &self_id,
+						 &payload->reply_blinding,
+						 &payload->reply_first_node,
+						 &payload->reply_path,
+						 &submsg)) {
+		log_broken(ld->log, "bad got_onionmsg_tous: %s",
```

Comment:
nit: `got_onionmsg_tous` -> `got_onionmsg_to_us`

Reply:
Oops, this was actually in 2/3 and I already merged that.  Oh well, this code is obsolete anyway....